include/jemalloc/internal/hpa_opts.h

```diff
@@ -7,8 +7,53 @@
 /*
  * This file is morally part of hpa.h, but is split out for header-ordering
  * reasons.
+ *
+ * All of these hpa_shard_opts below are experimental. We are exploring more
+ * efficient packing, hugifying, and purging approaches to make efficient
+ * trade-offs between CPU, memory, latency, and usability. This means all of
+ * them are at the risk of being deprecated and corresponding configurations
+ * should be updated once the final version settles.
  */
 
+/*
+ * This enum controls when jemalloc hugifies/dehugifies pages.  Each mode may be
+ * more suitable depending on deployment environments.
```

Comment:
Just a subtlety... it might read better to state this as "how" rather than "when".  "when" might make more sense if all options were in terms of time, like "never" instead of "none".